vm/entrypoint/src/syscalls/host.rs

```diff
@@ -41,57 +42,54 @@ pub extern "C" fn call(
 }
 
 /// Read from host storage at the given address and key.
-///
-/// The output is stored in the `key` pointer.
 #[allow(unused_variables)]
-#[no_mangle]
-pub extern "C" fn read_storage(key: *mut u32) {
+pub fn read_storage(key: &[u32; 8]) -> [u32; 8] {
   #[cfg(target_os = "zkvm")]
   unsafe {
+    let mut result = *key;
     asm!(
         "ecall",
         in("t0") crate::syscalls::HOST_READ,
-        in("a0") key,
-    )
+        in("a0") result.as_mut_ptr(),
+    );
+    return result;
   }
 
   #[cfg(not(target_os = "zkvm"))]
   unreachable!()
 }
 
 /// Write to host storage at the given address and key.
-///
-/// The result status code is stored in the `key` pointer.
 #[allow(unused_variables)]
-#[no_mangle]
-pub extern "C" fn write_storage(key: *mut u32, value: *const u32) {
+pub fn write_storage(key: &[u32; 8], value: &[u32; 8]) -> StorageStatus {
   #[cfg(target_os = "zkvm")]
   unsafe {
+    let status: u32;
     asm!(
         "ecall",
         in("t0") crate::syscalls::HOST_WRITE,
-        in("a0") key,
-        in("a1") value,
-    )
+        in("a0") key.as_ptr(),
+        in("a1") value.as_ptr(),
+        lateout("t0") status,
```

Comment:
same question as in `get_balance`, how do we explicitly ensure we're reading LE here?